### PR TITLE
fix some pre-v3 function syntax

### PR DESCRIPTION
**Pull request by ccmtaylor**
please note that I've not written any reasonml yet (or OCaml for that matter), so please double-check if I got it right.

### CODE REVIEW

File: src/pages/guide/language/function.md [hunk 1]
```diff
@@ -204,7 +204,7 @@ let twelve = addFive(7);
 Actually, the above `add` is nothing but syntactic sugar for this:
 
 ```reason
-let add = (x, y) => x + y;
+let add = (x) => (y) => x + y;
```
Comment (zploskey) [inline src/pages/guide/language/function.md]: This was actually correct as it was. Your change with `refmt` to what was already there on reason 3.

Comment (ccmtaylor) [reply src/pages/guide/language/function.md]: hmm, I thought this paragraph was explaining that the parameter-list syntax is equivalent to the curried syntax. 

> Reason functions can automatically be **partially** called:
 
>  ```reason
>  let add = (x, y) => x + y;
>  let addFive = add(5);
>  let eleven = addFive(6);
>  let twelve = addFive(7);
>  ```
 
>   Actually, the above `add` is nothing but syntactic sugar for this:
>   
>  ```reason
> let add = (x, y) => x + y;
> ```

note that the definitions of `add` are identical, so the paragraph doesn't make much sense as-is.

Comment (zploskey) [reply src/pages/guide/language/function.md]: Aha, you're right, I didn't look at the wider context. That said, this all looks good to me.

**Comment by chenglou** [comment]
Hello! #222 is a superset of this. Symbolically merging this =)

**Pull request merged**